lib/Frontend/PrintingDiagnosticConsumer.cpp

```diff
@@ -69,6 +69,12 @@ void PrintingDiagnosticConsumer::handleDiagnostic(SourceManager &SM,
     return;
 
   printDiagnostic(SM, Info);
+  for (auto path : Info.EducationalNotePaths) {
+    auto buffer = llvm::MemoryBuffer::getFile(path);
```

Comment:
Can you go through the source manager’s virtual file system here?

Reply:
👍 , done